packages/wallet/src/services/types.ts

```diff
@@ -35,6 +36,27 @@ export interface UtxoTracker extends TransactionalTracker<Cardano.Utxo[]> {
   setUnspendable(utxo: Cardano.Utxo[]): Promise<void>;
 }
 
+/**
+ * The AddressDiscovery interface provides a mechanism to discover addresses in Hierarchical Deterministic (HD) wallets
+ * by performing a look-ahead search of a specified number of addresses.
+ */
+export interface AddressDiscovery {
+  /**
+   * This method performs a look-ahead search of 'n' addresses in the HD wallet using the chain history and
+   * the given key agent. The discovered addresses are returned as a list.
+   *
+   * @param keyAgent The key agent controlling the root key to be used to derive the addresses to be discovered.
+   * @param chainHistoryProvider A cardano chain history provider.
+   * @param lookAheadCount The number of addresses to look ahead in the search.
+   * @returns A promise that will be resolved into a GroupedAddress list containing the discovered addresses.
+   */
+  discover(
+    keyAgent: AsyncKeyAgent,
+    chainHistoryProvider: ChainHistoryProvider,
+    lookAheadCount: number
```

Comment:
Those are specific to HDSequentialDiscovery implementation. I think it's unlikely that it would be needed to change those dynamically, so I would put them to the constructor of `new HDSequentialDiscovery({ addressLookAheadSearch, chainHistoryProvider })` rather than as argument of discover
```suggestion
```

Reply:
In that case you could provide a new discovery strategy object? It just seems very specific to the HDSequentialDiscovery strategy, doesn't seem right on the AddressDiscovery interface.